src/runtime_src/core/common/xclbin_parser.cpp

```diff
@@ -393,13 +393,19 @@ get_softkernels(const axlf* top)
       auto begin = reinterpret_cast<const char*>(top) + pSection->m_sectionOffset;
       auto soft = reinterpret_cast<const soft_kernel*>(begin);
 
-      softkernel_object sko;
+      sks.emplace_back(softkernel_object{});
+      auto sko = sks.back();
```

Comment:
This is a bug.  you will be modifying  an in scope copy of the sks.back().  You probably mean auto&.
Please create sko by value as before, then at end use sks.emplace_back(std::move(sko))